### PR TITLE
PHOENIX-6379 Implement a new Metric Type which will be used for Table…

### PR DESCRIPTION
**Pull request by vmeka2020**
Implement a new Metric Type which will be used for TableMetrics

### CODE REVIEW

File: phoenix-core/src/test/java/org/apache/phoenix/monitoring/PhoenixMetricImplTest.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    Tests the functionality of the TableMetricImpl methods which are exposed
+    1. change()
+    2. increment()
+    3. decrement()
+    4. reset()
+    5. getValue()
+     */
+    @Test public void testTableMetricImplAvailableMethods() {
+
+        PhoenixMetric metric = new PhoenixMetricImpl(MetricType.SELECT_SQL_COUNTER);
```
Comment (yanxinyi) [inline phoenix-core/src/test/java/org/apache/phoenix/monitoring/PhoenixMetricImplTest.java:36]: I didn’t see the MetircType class

Comment (vmeka2020) [reply phoenix-core/src/test/java/org/apache/phoenix/monitoring/PhoenixMetricImplTest.java:36]: its in https://github.com/vmeka2020/phoenix/blob/PHOENIX-6379/phoenix-core/src/main/java/org/apache/phoenix/monitoring/MetricType.java

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 30s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ 4.x Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  14m  6s |  4.x passed  |
| +1 :green_heart: |  compile  |   1m  1s |  4.x passed  |
| +1 :green_heart: |  checkstyle  |   0m 26s |  4.x passed  |
| +1 :green_heart: |  javadoc  |   0m 43s |  4.x passed  |
| +0 :ok: |  spotbugs  |   3m  0s |  phoenix-core in 4.x has 945 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 45s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  1s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  1s |  the patch passed  |
| -1 :x: |  checkstyle  |   0m 27s |  phoenix-core: The patch generated 21 new + 0 unchanged - 0 fixed = 21 total (was 0)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 44s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   3m 10s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 131m 11s |  phoenix-core in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 12s |  The patch does not generate ASF License warnings.  |
|  |   | 163m 13s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1139/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1139 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile |
| uname | Linux a31723760615 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | 4.x / 883170e |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1139/1/artifact/yetus-general-check/output/diff-checkstyle-phoenix-core.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1139/1/testReport/ |
| Max. process+thread count | 5692 (vs. ulimit of 30000) |
| modules | C: phoenix-core U: phoenix-core |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1139/1/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: phoenix-core/src/main/java/org/apache/phoenix/monitoring/PhoenixMetric.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ */
+package org.apache.phoenix.monitoring;
+
+/**
+ * Class that exposes the various phoenix metrics collected
+ * at the Table level. Because metrics are dynamic in nature, it is not guaranteed that the
+ * state exposed will always be in sync with each other. One should use
+ * these metrics primarily for monitoring and debugging purposes.
+ */
+public interface PhoenixMetric extends Metric {
```
Comment (virajjasani) [inline phoenix-core/src/main/java/org/apache/phoenix/monitoring/PhoenixMetric.java:26]: `PhoenixMetric` seems quite generic. Let's make it `PhoenixTableMetric` and `PhoenixTableMetricImpl` ?

Comment (vmeka2020) [reply phoenix-core/src/main/java/org/apache/phoenix/monitoring/PhoenixMetric.java]: Modified it.

**Review by yanxinyi — APPROVED** [review]
Thanks for the PR, can you also raise a new PR for the master? I will merge it after the Hadoop QA run.

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 30s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ 4.x Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  13m 41s |  4.x passed  |
| +1 :green_heart: |  compile  |   0m 59s |  4.x passed  |
| +1 :green_heart: |  checkstyle  |   0m 27s |  4.x passed  |
| +1 :green_heart: |  javadoc  |   0m 46s |  4.x passed  |
| +0 :ok: |  spotbugs  |   2m 59s |  phoenix-core in 4.x has 945 extant spotbugs warnings.  |
| -0 :warning: |  patch  |   3m  6s |  Used diff version of patch file. Binary files and potentially other changes not applied. Please rebase and squash commits if necessary.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 44s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  0s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  0s |  the patch passed  |
| -1 :x: |  checkstyle  |   0m 26s |  phoenix-core: The patch generated 21 new + 0 unchanged - 0 fixed = 21 total (was 0)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 43s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   3m 13s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 129m 44s |  phoenix-core in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 10s |  The patch does not generate ASF License warnings.  |
|  |   | 161m 24s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1139/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1139 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile |
| uname | Linux c86e764651b2 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | 4.x / b626e35 |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1139/2/artifact/yetus-general-check/output/diff-checkstyle-phoenix-core.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1139/2/testReport/ |
| Max. process+thread count | 5816 (vs. ulimit of 30000) |
| modules | C: phoenix-core U: phoenix-core |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1139/2/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**